src/utils/codeowners.ts

```diff
@@ -63,17 +64,66 @@ const parseCodeOwner = (filePath: string, codeOwnerContent: string): ownerRule[]
   return filteredRules.map((rule) => ({ filePath, ...createMatcherCodeownersRule(filePath, rule) }));
 };
 
+const translateGlob = (glob: string) => {
+  const parsedGlob = parseGlob(glob);
+
+  // These controls are based on the Github CODEOWNERS syntax documentation
+  // https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/about-code-owners#codeowners-syntax
+  // as well as the gitignore pattern format which it extends
+  // https://git-scm.com/docs/gitignore#_pattern_format
+
+  if (parsedGlob.is.negated) {
+    throw new Error('Rule cannot contain negations');
+  }
+  if (parsedGlob.is.braces) {
+    throw new Error('Rule cannot contain braces');
+  }
+  if (glob.startsWith('\\#')) {
+    throw new Error('Rule cannot start with escaped #');
+  }
+  if (parsedGlob.is.glob && /\[(?:.-.)+\]/.test(parsedGlob.glob)) {
```

Comment:
can you make a const out of the regular expression and add some comment to it? I believe there's a constants file. 

regular expressions get confusing over time! so having them with words explaining what is trying to match is always  preferred

Reply:
This is fixed now. I called it `CHARACTER_RANGE_PATTERN` since it should match `[a-z]` patterns.